guides/20240819_history_embedding.md

```diff
@@ -0,0 +1,276 @@
+
+```bash
+daytona code
+```
+
+### Set Up OpenAI Service (Optional)
+
+There are many hosted models compatible with the OpenAI API. Here's a basic rundown of available services:
+
+|                             | **Azure**                   | **OpenAI**                   | **Cohere**                  | **Anthropic**                | **Jurassic-2**              |
+|-----------------------------|-----------------------------|------------------------------|-----------------------------|------------------------------|-----------------------------|
+| **Model Size**              | 175B+ parameters            | 1B-175B+ parameters          | 6B-12B parameters           | 100B+ parameters             | 20B-178B parameters         |
+| **Context Length**          | 4,000-32,000 tokens         | 4,000-32,000 tokens          | Up to 2,048 tokens          | Up to 100,000 tokens         | Up to 2,048 tokens          |
+| **Vector Size**             | 1,536 dimensions            | 1,536 dimensions             | 512-768 dimensions          | 768-1,024 dimensions         | 1,024-2,048 dimensions      |
+| **Time for Embedding**      | 0.5 - 6 seconds (varies)    | 0.5 - 6 seconds (varies)     | 0.2 - 4 seconds (varies)    | 0.5 - 6 seconds (varies)     | 1 - 5 seconds (varies)      |
+| **API Cost**                | $0.0004-$0.0120 per token   | $0.0004-$0.0120 per token    | $0.0001-$0.0005 per token   | $0.0005-$0.0030 per token    | $0.0005-$0.0020 per token   |
+
+#### Key Notes
+
+1. **Model Size:**
+  - **Azure & OpenAI:** The sizes refer to GPT-3 and GPT-4 models. GPT-3 has 175B parameters; GPT-4 is larger but the exact size is proprietary.
+  - **Anthropic:** Claude models are estimated to be in the 100B+ range, but exact figures are not officially confirmed.
+
+2. **Context Length:**
+  - **OpenAI and Anthropic:** Both platforms offer models with larger context lengths, particularly in newer versions.
+  - **Cohere & Jurassic-2:** Smaller context lengths compared to OpenAI's latest models.
+
+3. **Time for Embedding:**
+  - This is highly variable and depends on multiple factors, such as the specific model, server load, and input length. The provided ranges should be taken as rough estimates rather than definitive benchmarks.
+
+4. **API Cost:**
+  - **Azure & OpenAI:** The cost can vary significantly based on the model, with GPT-4 generally costing more per token than GPT-3.
+  - **Cohere, Anthropic, and Jurassic-2:** These ranges are more general and reflect common usage scenarios. Always check the latest pricing from providers as it can change.
+
+#### Environment Variables
+
+Ensure you have the necessary environment variables configured in a local `.env` file. For example, this is how you would configure Azure:
+
+```env
+AZURE_API_VERSION=<your_azure_api_version>
+AZURE_ENDPOINT=<your_azure_endpoint>
+AZURE_OPENAI_API_KEY=<your_azure_api_key>
+```
+
+## Embedding and Searching Your Browser History
+
+### Embedding
+
```

Comment:
@giraffekey  you could add a step to tell users to run the command `pip install sentence_transformers` here.